subsys/net/lib/http/http_server_http1.c

```diff
@@ -443,7 +468,34 @@ int handle_http1_request(struct http_client_ctx *client)
 	}
 
 	if (client->has_upgrade_header) {
-		return handle_http1_to_http2_upgrade(client);
+		static const char upgrade_required[] =
+			"HTTP/1.1 426 Upgrade required\r\n"
+			"Upgrade: ";
+		static const char upgrade_msg[] =
+			"Content-Length: 13\r\n\r\n"
+			"Wrong upgrade";
+		char *needed_upgrade = "h2c\r\n";
+
+		if (client->websocket_upgrade) {
+			goto upgrade_not_found;
+		}
+
+		if (client->http2_upgrade) {
+			return handle_http1_to_http2_upgrade(client);
+		}
+
+upgrade_not_found:
+		(void)zsock_send(client->fd, upgrade_required,
```

Comment:
Shouldn't we stick to `http_server_sendall()`? In theory, `zsock_send()` may not send al the requested data.

Reply:
I was using `zsock_send()` here intentionally so that we could avoid sending partial message, there is `http_server_sendall()` few lines below that will make sure the headers are then fully sent.

Reply:
Hmm, I just realized that I was wrong here, I need to change this to use sendall version.